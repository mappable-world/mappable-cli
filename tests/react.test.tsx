import React from 'react';
import ReactDOM from 'react-dom';
import {render} from '@testing-library/react';
import type {LngLat, MMap} from '@mappable-world/mappable-types';

describe('React MMap test', () => {
    const LOCATION = {center: [37.62, 55.75] as LngLat, zoom: 12};
    let container: HTMLElement;
    let map: MMap;

    beforeEach(() => {
        container = document.createElement('div');
        Object.assign(container.style, {width: '640px', height: '480px'});

        document.body.appendChild(container);
    });

    afterEach(() => {
        map?.destroy();
        container?.remove();
    });

    it('should make react map', async () => {
        expect(mappable).toBeDefined();

        const [mappableReact] = await Promise.all([
            mappable.import('@mappable-world/mappable-reactify'),
            mappable.ready
        ]);
        const reactify = mappableReact.reactify.bindTo(React, ReactDOM);
        expect(reactify).toBeDefined();

        const {MMap, MMapDefaultSchemeLayer, MMapDefaultFeaturesLayer, MMapControls, MMapControl} =
            reactify.module(mappable);
        expect(MMap).toBeDefined();

        render(
            <MMap location={LOCATION} ref={(x) => (map = x)}>
                <MMapDefaultSchemeLayer />
                <MMapDefaultFeaturesLayer />

                <MMapControls position="left">
                    <MMapControl>
                        <div>test control</div>
                    </MMapControl>
                </MMapControls>
            </MMap>,
            {
                container
            }
        );

        const tree = domToJson(map.container);
        expect(tree).toMatchSnapshot();
    });
});
