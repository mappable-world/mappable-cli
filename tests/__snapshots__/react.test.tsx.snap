// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React MMap test should make react map 1`] = `
{
  "attributes": {
    "class": "mappable--map-container",
  },
  "children": [
    {
      "attributes": {
        "class": "mappable--map",
        "style": "width: 640px; height: 480px;",
      },
      "children": [
        {
          "attributes": {
            "class": "mappable--main-engine-container",
          },
          "children": [
            {
              "attributes": {},
              "children": [
                {
                  "attributes": {
                    "class": "mappable--layer mappable--tile-layer",
                    "style": "perspective: 895.6921938165306px; opacity: 1;",
                  },
                  "children": [
                    {
                      "attributes": {
                        "class": "mappable--tile-layer__container",
                        "data-z": "12",
                        "style": "transform: 
            rotateX(0.00000rad)
            translate3d(0px, 0px, 0px)
            rotateZ(0.00000rad) translate3d(0px, 0px, 0px)
        ;",
                      },
                      "nodeName": "mappable",
                    },
                  ],
                  "nodeName": "mappable",
                },
              ],
              "nodeName": "mappable",
            },
          ],
          "nodeName": "mappable",
        },
        {
          "attributes": {
            "class": "mappable--top-engine-container",
          },
          "children": [
            {
              "attributes": {},
              "children": [
                {
                  "attributes": {
                    "class": "mappable--layer mappable--graphics-layer",
                    "style": "perspective: 895.6921938165306px; opacity: 1; display: none;",
                  },
                  "children": [
                    {
                      "attributes": {
                        "xlink": "http://www.w3.org/1999/xlink",
                      },
                      "children": [
                        {
                          "attributes": {},
                          "nodeName": "g",
                        },
                      ],
                      "nodeName": "svg",
                    },
                  ],
                  "nodeName": "mappable",
                },
                {
                  "attributes": {
                    "class": "mappable--layer mappable--marker-layer",
                    "style": "opacity: 1; transform-origin: 320px 240px;",
                  },
                  "nodeName": "mappable",
                },
              ],
              "nodeName": "mappable",
            },
          ],
          "nodeName": "mappable",
        },
      ],
      "nodeName": "mappable",
    },
    {
      "attributes": {
        "class": "mappable--controls mappable--controls_center mappable--controls_left mappable--controls_vertical",
      },
      "children": [
        {
          "attributes": {
            "class": "mappable--control mappable--control__background",
          },
          "children": [
            {
              "attributes": {},
              "children": [
                "test control",
              ],
              "nodeName": "div",
            },
          ],
          "nodeName": "mappable",
        },
      ],
      "nodeName": "mappable",
    },
    {
      "attributes": {
        "class": "mappable--map-copyrights mappable--map-copyrights_bottom mappable--map-copyrights_right",
      },
      "children": [
        {
          "attributes": {
            "class": "mappable--map-copyrights__container",
          },
          "children": [
            {
              "attributes": {
                "class": "mappable--map-copyrights__text",
              },
              "nodeName": "span",
            },
            {
              "attributes": {
                "class": "mappable--map-copyrights__user-agreements",
                "href": "https://mappable.world/legal/tos",
                "target": "_blank",
              },
              "children": [
                "User Agreement",
              ],
              "nodeName": "a",
            },
          ],
          "nodeName": "div",
        },
        {
          "attributes": {
            "class": "mappable--icon-logo-en-black mappable--map-copyrights__logo",
            "style": "width: 80px; height: 20px;",
          },
          "nodeName": "span",
        },
        {
          "attributes": {
            "class": "mappable--map-copyrights__scale",
          },
          "nodeName": "div",
        },
      ],
      "nodeName": "div",
    },
  ],
  "nodeName": "mappable",
}
`;
